identity/legacy.go

```diff
@@ -23,11 +23,15 @@ func (l *Legacy) Auth(creds *Credentials) (*AuthDetails, error) {
 	request.Header.Set("X-Auth-User", creds.User)
 	request.Header.Set("X-Auth-Key", creds.Secrets)
 	response, err := l.client.Do(request)
-	defer response.Body.Close()
+	defer func() {
```

Comment:
I'll try. I ran into cases where this would cause panic because something was nil and dereferenced.